build_tools/cmake/iree_macros.cmake

```diff
@@ -76,10 +76,18 @@ function(iree_package_ns PACKAGE_NS)
   elseif(_RELATIVE_PATH MATCHES "^runtime/src$")
     # runtime/src (defs) -> iree::runtime::defs
     set(_PACKAGE "iree::runtime")
+  elseif(_RELATIVE_PATH MATCHES "^runtime/bindings/(.*)")
+    set(_PACKAGE "iree::bindings::${CMAKE_MATCH_1}")
+  elseif(_RELATIVE_PATH MATCHES "^build_tools/(.*)")
+    set(_PACKAGE "iree::build_tools::${CMAKE_MATCH_1}")
```

Comment:
:fearful: this whole setup skeeves me out. We used to have package name directly derivable from directory path and now we've got this complicated regex formula

Reply:
Yeah... I think remapping the targets is worth doing, but the way this is implemented also concerns me.

* Organized package names help when using IREE alongside other CMake projects
  * MLIR uses `MLIRLibraryName` and `mlir-tool-name`, LLVM uses `LLVMLibraryName` and `llvm-tool-name`
  * Some of the mess can be hidden downstream with `-DIREE_BUILD_TESTS=OFF` (for example, we set `CPUINFO_BUILD_UNIT_TESTS`, `BENCHMARK_ENABLE_TESTING`, `LLVM_INCLUDE_TESTS`, `SPIRV_HEADERS_SKIP_EXAMPLES`, and a few other options when we include other projects - projects downstream from IREE would do the same for us)
* Having predictable package names (matching the directory structure exactly) is useful. We _could_ just prefix with `iree_`, instead of doing this trimming, maybe keeping the exceptions for `runtime/src/` and `compiler/src`?
* The general style of regex replacement is brittle. We could try refactoring or adding more comments.

(I'm also considering splitting the target namespace mappings into their own PR)

Reply:
Yeah this can be a separate PR, but I think something that is at least a standard formula that doesn't require explicitly listing all our directories would be good. Something like directories up to one called "src" would be reasonable, for instance